app/controllers/admin/enterprises_controller.rb

```diff
@@ -326,9 +325,8 @@ def strip_new_properties
                                                      :producer_properties_attributes).nil?
         names = Spree::Property.pluck(:name)
         enterprise_params[:producer_properties_attributes].each do |key, property|
-          unless names.include? property[:property_name]
-            enterprise_params[:producer_properties_attributes].delete key
-          end
+          enterprise_params[:producer_properties_attributes].delete key unless
+            names.include? property[:property_name]
```

Comment:
At this point, I'm not sure how this came to be. I'm going to chart it up to a mistake on my part and I need to make smaller commits/PRs so that it is easier to keep track of everything. Lesson learned. I'm going to revert it.